src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -348,10 +348,15 @@ public void notifyOfNewStep(@NonNull Step step, @NonNull StepContext context) {
                 return;
             }
             if (computer.getAction(OpenTelemetryAttributesAction.class) == null) {
-                LOGGER.log(Level.WARNING, "Unexpected missing " + OpenTelemetryAttributesAction.class + " on " + computer + " fallback");
+                LOGGER.log(Level.WARNING, "Unexpected missing " + OpenTelemetryAttributesAction.class + " on " + computer + ", adding fallback");
                 String hostName = computer.getHostName();
                 OpenTelemetryAttributesAction openTelemetryAttributesAction = new OpenTelemetryAttributesAction();
-                openTelemetryAttributesAction.getAttributes().put(ResourceAttributes.HOST_NAME, hostName);
+                if (hostName != null) {
+                    // getHostName() returns null if the master cannot find the host name, e.g. due to network settings.
+                    // @see hudson.model.Computer#getHostName()
+                    openTelemetryAttributesAction.getAttributes().put(ResourceAttributes.HOST_NAME, hostName);
```

Comment:
Worth to inform the user that the host configuration is not correct with a log message

Reply:
That would be a lot of warnings! This code only executes if there is no `OpenTelemetryAttributesAction` on the `Computer`, and the only function that ever adds one is `MonitoringComputerListener`. So, if the listener is never called or fails to add the action, there will be a warning for every single stage/etc. on that Computer.

Also, it's not necessarily a misconfiguration if `getHostName` returns `null`, it just means Jenkins was not able to find a name. Since Jenkins doesn't treat that as an error, I think this plugin should also not treat it as an error.